astropy/coordinates/baseframe.py

```diff
@@ -275,22 +273,20 @@ def __init_subclass__(cls, **kwargs):
         cls._create_readonly_property('frame_specific_representation_info',
                                       copy.deepcopy(repr_info))
 
-        # Set the frame attributes
+        # Set the frame attributes. We follow frames in reverse order to keep
+        # insertion order and also allow subclasses to override attributes
+        # already set by their parent while keeping order.
         # TODO: Should this be made to use readonly_prop_factory as well or
         # would it be inconvenient for getting the frame_attributes from
         # classes?
-        if frame_attrs is None:
-            frame_attrs = {}
-        else:
-            frame_attrs = frame_attrs.copy()
-
-        for basecls in (cls,) + cls.__bases__:
+        frame_attrs = {}
```

Comment:
@embray - please check you agree with this change in logic. I got failures because any new `SkyOffsetFrame` class overrides `SkyOffsetFrame.origin` and then it either was not properly inserted in `frame_attributes` or, if I reset that one (first commit), it changed the order. I think not relying on a previous calculation of `frame_attributes` is probably better practice anyway.

Reply:
I admit I don't really understand the problem here since I don't know what a SkyOffsetFrame is or why it was broken.  What you did here makes sense and is actually close to what I think was my original approach to the problem.  I just didn't see that there was a difference and was trying to avoid two loops.  It didn't occur to me that there could be base classes with metaclasses that would do something different with frame_attributes.

IIUC that's what the problem was here, in which case your solution makes sense.